prov/efa/src/rdm/efa_rdm_rma.c

```diff
@@ -628,7 +650,10 @@ ssize_t efa_rdm_rma_writedata(struct fid_ep *ep, const void *buf, size_t len,
 	msg.rma_iov_count = 1;
 	msg.data = data;
 
-	return efa_rdm_rma_writemsg(ep, &msg, FI_REMOTE_CQ_DATA);
+	err = efa_rdm_generic_writemsg(efa_rdm_ep, &msg, FI_REMOTE_CQ_DATA);
```

Comment:
If you put the efa_perf_start to the generic one, you can get rid of the goto. Same comments for other places

Reply:
I want the efa_perf_start to start as soon as possible so we can capture as much of our cost as possible.

Reply:
This finally increase the instruction count for the shm path. Before it was just a return, now it is a `err = ..` and `return err`

Reply:
I'd be more concerned with how this kind of refactor affects the likelihood of tail call optimization/elimination for this function. @a-szegel do these changes improve performance or is it just speculative?

Reply:
I am removing duplicate calls to:
1. check rma capabilities
2. sync memops
3. getting the peer
4. seeing if we can use SHM

This should remove the amount of instructions in the critical path for EFA, which should help make us faster (less code is better than more code). 

I also fix our lttng code to make us able to capture the SHM cost, but this in turn gives us a goto out statement.  I can move the lttng code into the generic function as Shi suggests, but then we won't capture the SHM code.  I can run fi_rma_write tests with SHM before and after this change to confirm if I see a performance decrement via SHM.

Reply:
I do see a small perf hit for SHM... will move the lttng code inside the generic function.

Reply:
restored performance by removing goto + lttng macro.